monkey/monkey_island/cc/resources/auth/register.py

```diff
@@ -23,20 +24,24 @@ def __init__(self, authentication_service: AuthenticationService):
 
     def post(self):
         """
-        Registers a new user
+        Registers a new user using flask security register
 
-        Gets a username and password from the request sent from the client,
-        and registers a new user
-
-        :raises InvalidRegistrationCredentialsError: If username or password is empty
-        :raises AlreadyRegisteredError: If a user has already been registered
         """
-
-        username, password = get_username_password_from_request(request)
-
         try:
-            self._authentication_service.register_new_user(username, password)
-            return make_response({"error": ""}, HTTPStatus.OK)
-        # API Spec: HTTP status code for AlreadyRegisteredError should be 409 (CONFLICT)
-        except (InvalidRegistrationCredentialsError, AlreadyRegisteredError) as e:
-            return make_response({"error": str(e)}, HTTPStatus.BAD_REQUEST)
+            username, password = get_username_password_from_request(request)
+            if not username or not password:
+                return make_response(
+                    {"error": "Provided empty credentials"}, HTTPStatus.BAD_REQUEST
+                )
```

Comment:
Why isn't flask-security-too handling this?

Reply:
Flask-security-too is handling this but I think that we don't want to do a registration request at all if we provide empty credentials.

Reply:
How does Flask-Security-Too handle it?

Reply:
Why? Validating credentials is the responsibility of flask-security-too.

Reply:
We're using a third-party package to avoid exactly this. What's the point if we're going to re-implement functionality that the package is meant for?

Reply:
Good point. I will remove it.